### PR TITLE
Correct failing windows tests

### PR DESCRIPTION
**Pull request by IvanKolesarEquinor**
Set of commits to support windows tests and CI

### CODE REVIEW

File: xun/functions/store/sftp.py [hunk 1]
```diff
@@ -84,11 +84,10 @@ def ssh(self):
     def sftp(self):
         if self._sftp is None:
             self._sftp = self.ssh.open_sftp()
-            self._sftp.chdir(str(self.root))
             if not self.is_dir(self.root / 'values'):
-                self._sftp.mkdir('values', mode=0o711)
+                self._sftp.mkdir(str(self.root / 'values'), mode=0o711)
             if not self.is_dir(self.root / 'keys'):
-                self._sftp.mkdir('keys', mode=0o711)
+                self._sftp.mkdir(str(self.root / 'keys'), mode=0o711)
```
Comment (JensGM) [inline xun/functions/store/sftp.py:90]: This seems wrong? 🤔

File: xun/tests/test_dask_driver.py [hunk 2]
```diff
@@ -14,3 +14,4 @@ def test_dask_driver():
         result = blueprint.run(driver=dask_driver, store=store)
 
     assert result == expected
+    client.close(0.1)
```
Comment (JensGM) [inline xun/tests/test_dask_driver.py:17]: 0.1?

File: xun/tests/test_stores.py [hunk 3]
```diff
@@ -114,6 +116,8 @@ def test_store_is_collection(cls):
         assert store['c'] == 2
 
         assert set(store.keys()) >= {'a', 'b', 'c'}
+        if  hasattr(store.driver, 'cache'): store.driver.cache.close()
```
Comment (JensGM) [inline xun/tests/test_stores.py:119]: This should not be needed, and if confusing

File: xun/tests/test_stores.py [hunk 4]
```diff
@@ -127,6 +131,7 @@ def test_store_driver_is_collection(cls):
         assert driver['c'] == 2
 
         assert set(driver.keys()) >= {'a', 'b', 'c'}
+        if  hasattr(store.driver, 'cache'): store.driver.cache.close()
```
Comment (JensGM) [inline xun/tests/test_stores.py:134]: Goes for all of them

**Review by JensGM — CHANGES_REQUESTED** [review]
The cleanup seems to be a bit over the top, I think we can skip some of those steps

File: xun/functions/store/sftp.py (hunk 1 above)
Comment (IvanKolesarEquinor) [reply xun/functions/store/sftp.py:90]: It's the result of Paramiko nor really supporting Win OS in general. It only checks if working directory is already set in unix-based paths. Which means it checks if path starts with '/'. That is not present in Win based paths. In short by setting chdir first and then calling mkdir, on windows size the path created is something like: 'C:\...C:\....\values'.

This way we are on safe ground on multiple OS

File: xun/tests/test_dask_driver.py (hunk 2 above)
Comment (IvanKolesarEquinor) [reply xun/tests/test_dask_driver.py:17]: It worked with just closing. I have just read a recommendation to put there some small time offset just to be sure.

File: xun/tests/test_stores.py (hunk 3 above)
Comment (IvanKolesarEquinor) [reply xun/tests/test_stores.py:119]: Without additional closing of the DiskCache, the temporary cache database file (cache.db) is still taken and windows is failing on deleting the file.
Actually, this is exactly the part of the solution I'm unsure about.
And it's not confusing :) It just says to close cache if it's present in your driver :)

File: xun/tests/test_stores.py (hunk 4 above)
Comment (IvanKolesarEquinor) [reply xun/tests/test_stores.py:134]: The same as before.
I'm not sure how to tackle the problem differently. Are there any suggestions?

File: xun/functions/store/sftp.py (hunk 1 above)
Comment (JensGM) [reply xun/functions/store/sftp.py:90]: This is fine

File: xun/tests/test_stores.py [hunk 5]
```diff
@@ -26,7 +26,9 @@ def Memory():
 @contextmanager
 def TmpDiskCache():
     with tempfile.TemporaryDirectory() as tmpdirname:
-        yield xun.functions.store.DiskCache(tmpdirname)
+        ctx = xun.functions.store.DiskCache(tmpdirname)
+        yield ctx
+        ctx.driver.cache.close()
```
Comment (JensGM) [inline xun/tests/test_stores.py:31]: Delete DiskCache, remember to remove the dependency too :D

File: xun/tests/test_stores.py (hunk 3 above)
Comment (IvanKolesarEquinor) [reply xun/tests/test_stores.py]: DiskCache removed

File: xun/tests/test_stores.py (hunk 4 above)
Comment (IvanKolesarEquinor) [reply xun/tests/test_stores.py]: DiskCache removed

File: xun/tests/test_stores.py (hunk 5 above)
Comment (IvanKolesarEquinor) [reply xun/tests/test_stores.py]: DiskCache removed, also from dependencies :)

File: xun/tests/test_dask_driver.py (hunk 2 above)
Comment (IvanKolesarEquinor) [reply xun/tests/test_dask_driver.py]: done

**Review by JensGM — APPROVED** [review]

**Pull request merged**